src/infra/docs/bors/queue-resync.md

```diff
@@ -0,0 +1,131 @@
+# Fixing inconsistencies in the bors queue
+
+bors queue page: <https://bors.rust-lang.org/queue/rust>.
+
+<div class="warning">
+**WARNING**: This is a **destructive** operation. If someone syncs, they need to
+baby-sit the queue for around 45 minutes: around 30 minutes to wait for PRs to
+be recollected, and 15 minutes after that to kick out PRs that should not be in
+the tree.
+
+**DO NOT CLICK THIS BUTTON IF YOU ARE NOT ABLE TO HANDLE THE CLEANUP.**
+</div>
+
+Sometimes you have to do a bors queue sync for various reasons. This is not
+trivial and requires you to be very careful, as otherwise we may accidentally
+merge PRs to `master` (or even `beta`) that should not have been merged
+otherwise.
+
+**You can only do this if you have bors `r+` permissions on the rust-lang/rust
```

Comment:
Just drive-by comment: Does the "can" mean to "are able to" or "are allowed to"? I never pressed synchronize before I got granted r+ perms but I know that people without r+ can create rollups via that interface assuming they have connected their GitHub account to bors (or whichever app, I don't remember).

Reply:
You *can* but you really should not (there's no permission control on this), because if you don't have `r+` you can't fix the sync problems.

Reply:
> I know that people without r+ can create rollups via that interface assuming they have connected their GitHub account to bors (or whichever app, I don't remember). They can't r+ them ofc.

In general, I think it's intended that you should only create rollups if you have `r+` as well.

Reply:
I'm pretty sure syncs check github auth

Reply:
I never actually tried, if that's the case then nevermind, as the sentence written as-is would be accurate. @pietroalbini do you know if this is the case?

Reply:
Well sure GH auth but not bors privileges, right? Confer rollup PRs created by bors-unprivileged people: [is:pr in:title rollup of -label:rollup commenter:rustbot is:unmerged is:closed ](https://github.com/rust-lang/rust/pulls?q=is%3Apr+in%3Atitle+rollup+of+-label%3Arollup+commenter%3Arustbot+is%3Aunmerged+is%3Aclosed)